### DISCUSSION TITLE
Empty password fails login

### DISCUSSION THREAD

**Discussion by igh-gr** [neutrinolabs/xrdp#2474] [Q&A]
Hello,
I have a server that runs xrdp and a client that uses xfreerdp to connect. I set an empty password on my client and try to login. Sadly the login then fails, with "login failed for user…"

When I try the same to a windows RDP server I simply get the login screen and have to enter a password. 

Is this by design or is there any way to change that behavior ?

**Comment by matt335672** [comment]
The RDP client is supposed to request this behaviour in the [TS_INFO PDU](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/732394f5-e2b5-4ac5-8a0a-35345386b0d1) by setting the INFO_AUTOLOGIN field.  This wouldn't however be the first time we've seen a buggy client.

All this is assuming you've just got a vanilla username here, containing no non-printing characters. Usernames containing ASCII 0x1f are treated specially - see #1668.

At the moment, there's no way to change the behaviour. We don't enforce password policy in XRDP, as this is generally done at lower levels (e.g. by PAM), and what you're describing is perfectly valid as far as the protocol goes. On the other hand, it's a pretty extreme use-case, so to my mind we should probably add a config variable (default to false) to allow empty autologon passwords.

@metalefty - what do you think?

**Comment by metalefty** [comment]
Users with an empty password can not login with autologin credentials but can login with xrdp's login dialog. What about just declaring a session without a password field?  Users must click "OK" but just need to click "OK".

Adding a config variable to allow empty autologon passwords is not a bad idea but login dialog with no password field also works with no code changes. I think it is not adequate to make code changes for such an extreme use case.

```
[Xorg]
name=EmptyPassword
lib=libxup.so
username=ask
port=-1
code=20
```

This worked for me.

```
$ xfreerdp /v:xrdp.example.com /u:emptypassworduser
```

![image](https://user-images.githubusercontent.com/941609/208111348-881383b4-2d86-4b7d-8189-9c86f024722d.png)

**Comment by matt335672** [comment]
Excellent - thanks.

So if the client sends an empty password and has the autologin flag set, I think we should ignore the autologin flag. Do you agree?

**Reply by metalefty** [reply]
I agree.

**Reply by matt335672** [reply]
I'll get a PR put together in the near future. I'm doing a bit of catch-up at the moment.
